nucliadb/nucliadb/ingest/orm/knowledgebox.py

```diff
@@ -366,45 +364,19 @@ async def del_labelset(self, id: str):
 
     # Entities
     async def get_entities(self, entities: GetEntitiesResponse):
-        entities_key = KB_ENTITIES.format(kbid=self.kbid)
-        async for key in self.txn.keys(entities_key, count=-1):
-            entitygroup = await self.txn.get(key)
-            id = key.split("/")[-1]
-            if entitygroup is not None:
-                eg = EntitiesGroup()
-                eg.ParseFromString(entitygroup)
-                entities.groups[id].CopyFrom(eg)
-
-    async def get_entitiesgroup(
-        self, group: str, entitiesgroup: GetEntitiesGroupResponse
-    ):
-        payload = await self.get_entitiesgroup_inner(group)
-        if payload is not None:
-            entitiesgroup.group.ParseFromString(payload)
+        return await self.entities_manager.get_entities(entities)
 
-    async def get_entitiesgroup_inner(self, group: str):
-        entities_key = KB_ENTITIES_GROUP.format(kbid=self.kbid, id=group)
-        payload = await self.txn.get(entities_key)
-        return payload
+    async def get_entities_group(self, group: str) -> Optional[EntitiesGroup]:
+        return await self.entities_manager.get_entities_group(group)
 
     async def set_entities(self, group: str, entities: EntitiesGroup):
-        payload = await self.get_entitiesgroup_inner(group)
-        if payload is None:
-            eg = entities
-        else:
-            eg = EntitiesGroup()
-            eg.ParseFromString(payload)
-            eg.MergeFrom(entities)
-        entities_key = KB_ENTITIES_GROUP.format(kbid=self.kbid, id=group)
-        await self.txn.set(entities_key, eg.SerializeToString())
+        return await self.entities_manager.set_entities(group, entities)
```

Comment:
I couldn't find where you do the txn.rollback() in case there is an exception. 
I guess it should be done on the writer view?